WordPress/Classes/Extensions/MediaService+Local.swift

```diff
@@ -0,0 +1,72 @@
+import Foundation
+import ImageIO
+
+/// Encapsulates Media functions relative to the local Media directory.
+///
+extension MediaService {
+
+    fileprivate static let mediaDirectoryName = "Media"
+
+    /// Returns filesystem URL for the local Media directory.
+    ///
+    class func localMediaDirectory() throws -> URL {
+        let fileManager = FileManager.default
+        let documents = fileManager.urls(for: .documentDirectory, in: .userDomainMask).first!
```

Comment:
While I agree with all this suggestions for the media folder  to be changed to the caches directory, I will save those changes for a separate PR, with dedicated testing. 
My rational is that this PR already has lot of changes and is more focused on moving methods around and converting to Swift, so we should a more complex change to another PR.